### PR TITLE
feat(`cheatcodes`): add computeCreateAddress cheatcodes

### LINKED ISSUE #6270: feature(cheatcode): `vm.predictContractAddr()`; pre-determine next deployed addresses
### Component

Forge

### Describe the feature you would like

Would be useful to have a cheatcode that can pre-compute future contract deployment addresses based on deployer address and nonce:

```solidity
function predictContractAddr(address deployer, uint256 nonce) external returns (address);
```

Frequently run into use-cases where the contract ordering is dependent and I need to serialize some address/call for a future-deployed contract.

### Additional context

Today, I just copy-paste @k06a's handy [`addressFrom` function](https://ethereum.stackexchange.com/questions/24248/how-to-calculate-an-ethereum-contracts-address-during-its-creation-using-the-so) into a test-suite library.

```solidity
function addressFrom(address _origin, uint _nonce) public pure returns (address) {
    bytes memory data;
    if (_nonce == 0x00)          data = abi.encodePacked(bytes1(0xd6), bytes1(0x94), _origin, bytes1(0x80));
    else if (_nonce <= 0x7f)     data = abi.encodePacked(bytes1(0xd6), bytes1(0x94), _origin, uint8(_nonce));
    else if (_nonce <= 0xff)     data = abi.encodePacked(bytes1(0xd7), bytes1(0x94), _origin, bytes1(0x81), uint8(_nonce));
    else if (_nonce <= 0xffff)   data = abi.encodePacked(bytes1(0xd8), bytes1(0x94), _origin, bytes1(0x82), uint16(_nonce));
    else if (_nonce <= 0xffffff) data = abi.encodePacked(bytes1(0xd9), bytes1(0x94), _origin, bytes1(0x83), uint24(_nonce));
    else                         data = abi.encodePacked(bytes1(0xda), bytes1(0x94), _origin, bytes1(0x84), uint32(_nonce));
    return address(uint160(uint256(keccak256(data))));
}
```

### PR DESCRIPTION
**Pull request by qiweiii**
## Motivation

Closes #6270
add `computeCreateAddress` and `computeCreate2Address` to cheatcodes.

## Solution

As suggested in [comment](https://github.com/foundry-rs/foundry/issues/6270#issuecomment-1807187525), I transformed these code in [forge-std](https://github.com/foundry-rs/forge-std/blob/37a37ab73364d6644bfe11edf88a07880f99bd56/src/StdUtils.sol#L97-L134) to cheatcodes.

I am very new to rust, please correct me if any thing is wrong.

### CODE REVIEW

**Review by mattsse — CHANGES_REQUESTED** [review]
thanks,

sorry you had to port these from solidity -.-

File: crates/cheatcodes/src/utils.rs [hunk 1]
```diff
@@ -111,6 +112,59 @@ impl Cheatcode for getLabelCall {
...
+        let Self { nonce, deployer } = self;
+        if *nonce == Uint::from(0x00) {
+            return address_from_last_20_bytes(&keccak256(
+                &[&[0xd6], &[0x94], deployer.as_slice(), &[0x80]].abi_encode_packed(),
+            ));
+        }
+        if *nonce <= Uint::from(0x7f) {
+            return address_from_last_20_bytes(&keccak256(
+                &[&[0xd6], &[0x94], deployer.as_slice(), &nonce.to_be_bytes::<1>()]
+                    .abi_encode_packed(),
```
[the remark this answers is not in the document]
Comment (qiweiii) [reply crates/cheatcodes/src/utils.rs]: updated

File: crates/cheatcodes/src/utils.rs [hunk 2]
```diff
@@ -111,6 +112,27 @@ impl Cheatcode for getLabelCall {
     }
 }
 
+impl Cheatcode for computeCreateAddressCall {
+    fn apply(&self, _state: &mut Cheatcodes) -> Result {
+        let Self { nonce, deployer } = self;
+        Ok(deployer.create(nonce.to()).abi_encode())
```
Comment (DaniPopes) [inline crates/cheatcodes/src/utils.rs:118]: `.to()` panics if the number does not fit the target type (u64), can you add a check before this? You can copy it from `chainIdCall`

**Review by DaniPopes — CHANGES_REQUESTED** [review]

File: crates/cheatcodes/src/utils.rs (hunk 2 above)
Comment (qiweiii) [reply crates/cheatcodes/src/utils.rs:118]: sure

**Review by DaniPopes — APPROVED** [review]
Thanks!

**Pull request merged**